src/main/java/net/pms/dlna/DLNAMediaDatabase.java

```diff
@@ -1252,6 +1256,62 @@ public synchronized void cleanup() {
 					}
 				}
 			}
+
+			/**
+			 * Cleanup of THUMBNAILS table
+			 *
+			 * Removes entries that are not referenced by any rows in the FILES table.
+			 */
+			ps = conn.prepareStatement(
+				"DELETE FROM THUMBNAILS " +
+				"WHERE NOT EXISTS (" +
+					"SELECT ID FROM FILES " +
+					"WHERE FILES.THUMBID = THUMBNAILS.ID" +
+				");"
+			);
+			rs = ps.executeQuery();
+
+			/**
+			 * Cleanup of FILES_STATUS table
+			 *
+			 * Removes entries that are not referenced by any rows in the FILES table.
+			 */
+			ps = conn.prepareStatement(
+				"DELETE FROM FILES_STATUS " +
+				"WHERE NOT EXISTS (" +
+					"SELECT ID FROM FILES " +
+					"WHERE FILES.FILENAME = FILES_STATUS.FILENAME" +
+				");"
+			);
+			rs = ps.executeQuery();
+
+			/**
+			 * Cleanup of AUDIOTRACKS table
+			 *
+			 * Removes entries that are not referenced by any rows in the FILES table.
+			 */
+			ps = conn.prepareStatement(
+				"DELETE FROM AUDIOTRACKS " +
+				"WHERE NOT EXISTS (" +
+					"SELECT ID FROM FILES " +
+					"WHERE FILES.ID = AUDIOTRACKS.FILEID" +
+				");"
+			);
+			rs = ps.executeQuery();
+
+			/**
+			 * Cleanup of SUBTRACKS table
+			 *
+			 * Removes entries that are not referenced by any rows in the FILES table.
+			 */
+			ps = conn.prepareStatement(
+				"DELETE FROM SUBTRACKS " +
+				"WHERE NOT EXISTS (" +
+					"SELECT ID FROM FILES " +
+					"WHERE FILES.ID = SUBTRACKS.FILEID" +
+				");"
+			);
+			rs = ps.executeQuery();
```

Comment:
How about adding DELETE CASCADE constraint instead to child tables?

Reply:
That's a good idea, thanks. That should work for all of them except the THUMBNAILS table

Reply:
That's done now, not fully tested but is that what you meant?

Reply:
Yes, the constraints look fine.